### PR TITLE
#1_4Task.Max.Cichanau

### PR DESCRIPTION
**Pull request by Vpartyzan**
https://travis-ci.org/github/Vpartyzan/js-assignments

### CODE REVIEW

**Comment by Vpartyzan** [comment]
https://travis-ci.org/github/Vpartyzan/js-assignments

File: task/01-strings-tasks.js [hunk 1]
```diff
@@ -69,7 +69,7 @@ function getStringFromTemplate(firstName, lastName) {
  *   'Hello, Chuck Norris!' => 'Chuck Norris'
  */
 function extractNameFromTemplate(value) {
-    throw new Error('Not implemented');
+    return value.slice(7, (value.length - 1));
```
Comment (FedorovD) [inline task/01-strings-tasks.js]: looks good, but please note slice can work with negative numbers also to take symbols from end. In the case `slice(7, -1);` can do the same.

File: task/02-numbers-tasks.js [hunk 2]
```diff
@@ -221,7 +228,7 @@ function isPrime(n) {
  *   toNumber(new Number(42), 0) => 42
  */
 function toNumber(value, def) {
-    throw new Error('Not implemented');
+    return +value ? +value : def;
```
Comment (FedorovD) [inline task/02-numbers-tasks.js]: not correct, if value = '0' ?

**Comment by FedorovD** [comment]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/02-numbers-tasks.js [hunk 3]
```diff
@@ -221,7 +228,7 @@ function isPrime(n) {
  *   toNumber(new Number(42), 0) => 42
  */
 function toNumber(value, def) {
-    throw new Error('Not implemented');
+    return parseFloat(value) ? parseFloat(value) : def;
```
Comment (FedorovD) [inline task/02-numbers-tasks.js:231]: still not working, you need check is `value` a Number, `parseFloat('0')` is interpreted as false and return `def`
see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN

**Comment by FedorovD** [comment]
Closing the PR as accepted, but pay attention to the comment on the task `toNumber`. Thanks!

**Pull request closed without merging**